click_odoo_contrib/backupdb.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright 2018 ACSONE SA/NV (<http://acsone.eu>)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+import json
+import os
+import shutil
+import tempfile
+import zipfile
+
+import click
+import click_odoo
+from click_odoo import odoo
+
+from ._dbutils import db_exists
+
+
+def _dump_db(dbname, backup):
+    cmd = ["pg_dump", "--no-owner", "--format=c", dbname]
+    _stdin, stdout = odoo.tools.exec_pg_command_pipe(*cmd)
+    backup.write(stdout, "dump.sql")
+
+
+def _create_manifest(cr, dbname, backup):
+    with tempfile.NamedTemporaryFile(mode="w") as f:
+        json.dump(odoo.service.db.dump_db_manifest(cr), f, indent=4)
+        f.seek(0)
+        backup.addfile(f.name, "manifest.json")
+
+
+def _backup_filestore(dbname, backup):
+    filestore_source = odoo.tools.config.filestore(dbname)
+    if os.path.isdir(filestore_source):
+        backup.copytree(filestore_source)
+
+
+class _Backup(object):
```

Comment:
Nitpicking:

```python
class AbstractBackup
class ZipBackup
class FolderBackup

@contextmanager
def backup(format, path):
    if format == 'zip':
        backup ZipBackup(path)
    ...
    try:
        yield backup
        backup.close()
    except:
        backup.delete()
        raise
```

Reply:
indeed, indeed it will be cleaner

Reply:
Moreover, we can put these classes in a separate file/module to be used by the future `restoredb` command as well

Reply:
@sbidoul done